File: attempt1/rust-attempt/src/nnn/link.rs
use std::cell::RefCell;
use neuron::Neuron;

pub struct Link<'a> {
  pub input: &'a RefCell<Neuron<'a>>,
  pub output: &'a RefCell<Neuron<'a>>
}

impl<'a> Link<'a> {
  pub fn new(output: &'a RefCell<Neuron<'a>>, input: &'a RefCell<Neuron<'a>>)
      -> Link<'a> {
    Link {input: input, output: output}
  }
}

File: attempt1/go-attempt/src/link.go
package nnn

type Link struct {
	input  *Neuron
	output *Neuron
}

func NewLink(input *Neuron, output *Neuron) *Link {
	result := &Link{input, output}
	input.outputs = append(input.outputs, result)
	output.inputs = append(output.inputs, result)
	return result
}

func (self *Link) Remove() {
	removeFromList(&self.input.outputs)
	removeFromList(&self.output.inputs)
}

func (self *Link) removeFromList(list *[]*Link) {
	for i, x := range list {
		if x == self {
			list[i] = list[len(list)-1]
			*list = list[0 : len(list)-1]
		}
	}
}
